### ISSUE TITLE
How to build definitive guide

### ISSUE THREAD

**Issue by Vadi** [elastic/elasticsearch-definitive-guide#270]
Would love a README file that shows nicely about how to build this guide locally. 
Some one care help

**Comment by mcandre** [comment]
I'm trying to build the basic epub, with or without author metadata. But I get an XML error:

```
$ brew install asciidoc
$ a2x -d book -f epub book.asciidoc -v
...
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment.asciidoc
asciidoc: 07_Admin.asciidoc: line 29: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/10_dynamic_settings.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 9: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/10_dynamic_settings.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/20_logging.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 11: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/20_logging.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/30_indexing_perf.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 13: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/30_indexing_perf.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/40_rolling_restart.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 15: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/40_rolling_restart.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/50_backup.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 17: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/50_backup.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/60_restore.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 19: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/60_restore.asciidoc
asciidoc: include: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/70_conclusion.asciidoc
asciidoc: 520_Post_Deployment.asciidoc: line 21: reading: /Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/520_Post_Deployment/70_conclusion.asciidoc

a2x: executing: "xmllint" --nonet --noout --valid "/Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/book.xml"

I/O error : Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd
/Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/book.xml:2: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
D DocBook XML V4.5//EN" "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
                                                                               ^
/Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/book.xml:6: validity error : Validation failed: no DTD found !
<book lang="en">
               ^

a2x: ERROR: "xmllint" --nonet --noout --valid "/Users/apennebaker/Desktop/src/elasticsearch-definitive-guide/book.xml" returned non-zero exit status 4
```

System:

```
$ specs asciidoc brew os
Specs:

specs 0.12
https://github.com/mcandre/specs#readme

asciidoc --version
asciidoc 8.6.9

brew --version
0.9.5

system_profiler SPSoftwareDataType | grep 'System Version'
      System Version: OS X 10.10.1 (14B25)
```

**Comment by bchatard** [comment]
@mcandre in asciidoc install (`brew info asciidoc`), is mentioned to add this to *rc file `export XML_CATALOG_FILES=/usr/local/etc/xml/catalog`. Hope this solve your issue (for me it works).

**Comment by Pryz** [comment]
@bchatard it works for me. Thanks ;)

**Comment by abulhol** [comment]
Hi there, I have tried to build the book with a2x, but I get an error like this:
a2x: ERROR: missing resource: /home/ME/elasticsearch-definitive-guide/images/icons/callouts/1.png
(I cloned the master branch with git)

**Comment by raf64flo** [comment]
Hi, you may try fix proposed by @pmusa on : https://github.com/elastic/elasticsearch-definitive-guide/pull/321#commitcomment-9854899

**Comment by abulhol** [comment]
Cool, thanks a lot!

**Comment by rkalyankumar** [comment]
I have built successfully both epub & pdf formats of the book. I use ubuntu 14.04.3 LTS : 

-  Install following sofwares: sudo apt-get install asciidoc docbook2X fop
-  Create symbolic link: ln -sf <full path where the git repository is cloned on the local> images/icons
     for e.g. if you have cloned git repository to home directory the above command looks like this:
           ln -sf $HOME/elasticsearch-definitive-guide images/icons
     Reference for this step: https://github.com/elastic/elasticsearch-definitive-guide/pull/321#commitcomment-9854899
-  To build pdf run following command: a2x -d book -f pdf --fop book.asciidoc -v
-  To build epub run following command: a2x -d book -f epub book.asciidoc -v

**Comment by xamox** [comment]
@rkalyankumar Thank you, your instructions worked for me.  Not sure why this isn't just in a README file in the repo.

**Comment by fractalf** [comment]
@rkalyankumar You are a hero and deserves a medal, thanks a lot! Been struggling for a couple of hours trying to build an epub of the docs, as I just bought a Kindle for the sole purpose of reading about elastic in bed.

..didn't understand (or do) that part where you symlink images/icons though, but the epub works fine on my Kindle now anyways :)

**Comment by polyfractal** [comment]
> Not sure why this isn't just in a README file in the repo.

The answer is basically "because the authors haven't spent time figuring out how to build the PDF"

But since @rkalyankumar did all the hard work, I'll add a note to the README when I get a chance (may be delayed, on a work trip atm and time is scarce).  Thanks @rkalyankumar for doing the work to figure it out! :)

**Comment by rkalyankumar** [comment]
My contribution is little as compared to the wealth of information that is there in those elasticsearch documentation! 

Thanks,
Kalyan

**Comment by polyfractal** [comment]
<3

If you want to, I'd be happy to merge a PR to the readme, since I won't have time to get to this until sometime next week (yay work trips :) )

**Comment by rkalyankumar** [comment]
Sure thing. What's the process? Should I fork pull repo?

**Comment by raf64flo** [comment]
I already created a PR few time ago: https://github.com/raf64flo/elasticsearch-definitive-guide/commit/67b3fa5e4cd2f7cbf82d6d56df638227d9f69fe2
But perhaps it does not the trick since it is kind of unrefined.
If you want I'll take some minutes to update it with @rkalyankumar 's proposals. Or may you fork it directly to fix it if you mind. :)

**Comment by raf64flo** [comment]
Just have updated https://github.com/elastic/elasticsearch-definitive-guide/pull/321
Tests and review welcome.

**Comment by danielmitterdorfer** [comment]
Closed by 7fad216f895b3e69dd8312bf401b634b7ba2c4ce
